File: src/test/java/kitchenpos/menus/tobe/domain/MenuProductTest.java
package kitchenpos.menus.tobe.domain;

import static java.math.BigDecimal.valueOf;
import kitchenpos.menus.tobe.infra.DefaultProfanities;
import kitchenpos.menus.tobe.domain.exception.InvalidMenuProductQuantityException;
import kitchenpos.menus.tobe.domain.vo.Profanities;
import kitchenpos.products.tobe.domain.Product;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

import java.util.UUID;

class MenuProductTest {

    @ParameterizedTest
    @ValueSource(ints = {-1, -100, -99999})
    void 메뉴에_등록된_상품의_수량은_0개_이상이어야_한다(int invalidQuantity) {
        // given
        Product product = new Product("후라이드치킨", valueOf(20_000));

        // when & then
        Assertions.assertThatThrownBy(() -> new MenuProduct(product, 20_000, invalidQuantity, UUID.randomUUID()))
                .isInstanceOf(InvalidMenuProductQuantityException.class)
                .hasMessage("메뉴에 등록된 상품의 수량은 0개 이상이어야 합니다.");
    }
}

File: src/test/java/kitchenpos/menus/tobe/domain/MenuProductTest.kt
package kitchenpos.menus.tobe.domain

import io.kotest.assertions.throwables.shouldThrow
import io.kotest.core.spec.style.BehaviorSpec
import io.kotest.matchers.shouldBe
import java.util.UUID

private val DEFAULT_PRODUCT_ID = UUID.randomUUID()

class MenuProductTest : BehaviorSpec({
    Given("메뉴 상품을 생성할 때") {
        When("메뉴 상품의 수량이 0 이하인 경우") {
            val invalidQuantity = 0

            Then("예외가 발생한다") {
                shouldThrow<IllegalArgumentException> {
                    MenuProduct(invalidQuantity, DEFAULT_PRODUCT_ID, 1000)
                }
            }
        }

        When("메뉴 상품의 수량이 1 이상인 경우") {
            val validQuantity = 1

            Then("메뉴 상품이 생성된다") {
                val menuProduct = MenuProduct(validQuantity, DEFAULT_PRODUCT_ID, 1000)

                menuProduct.productId shouldBe DEFAULT_PRODUCT_ID
                menuProduct.quantity shouldBe 1
                menuProduct.productPrice shouldBe 1000
            }
        }
    }
})
